src/app/zap-templates/common/ClustersHelper.js

```diff
@@ -477,6 +477,11 @@ function asPromise(promise)
   const fn = pkgId => Clusters.init(this, pkgId).then(() => promise);
   return templateUtil.ensureZclPackageId(this).then(fn).catch(err => { console.log(err); throw err; });
 }
+function initClusters()
```

Comment:
So a few things:

1.  As a followup, please change asBlocks/asPromise to call initClusters, something like this:
    ```
    function asPromise(promise)
    {
        let init = initClusters.call(this);
        return init.then(() => promise);
    }
    ```
    or even simpler with async function:
    ```
    async function asPromise(promise)
    {
        await initClusters.call(this);
        return promise;
    }
    ```
    and something similar for asBlocks.
2. The places like `Clusters.getClusters` that just use `this.ready.then()`... they can't call `this.init` because they don't necessarily have the right package id, because their `this` is whatever.  But they _could_ call some function to get `ready` which checks that `this.ready.running` is true and logs + throws if not.  I just tried that now and I get nice generation errors under ClusterTestGeneration instead of silent incorrect behavior....  Maybe replace all the `this.ready` bits with `this.assertReady()` where `assertReady` looks like this:
    ```
    if (!this.ready.running) {
        let err = new Error("Clusters not inited");
        console.log("Not Inited: " + err.stack);
        throw err;
    }
    return this.ready;
    ```

Followup OK for item 2 as well, I guess, but please make sure it happens.

Reply:
As discussed, I will create a follow up PR to address these suggested changes.

Reply:
See https://github.com/project-chip/connectedhomeip/pull/12914